internal/storage/storage.go

```diff
@@ -18,12 +18,37 @@ var ErrAddressIsTaken = fmt.Errorf("address is taken")
 
 // Request ...
 type Request struct {
-	Owner       string       `db:"owner"`
-	Email       string       `db:"email"`
-	Address     string       `db:"address"`
-	Code        string       `db:"code"`
-	CreatedAt   time.Time    `db:"created_at"`
-	ConfirmedAt sql.NullTime `db:"confirmed_at"`
+	Owner        string       `db:"owner"`
+	Email        string       `db:"email"`
+	Address      string       `db:"address"`
+	Code         string       `db:"code"`
+	CreatedAt    time.Time    `db:"created_at"`
+	ConfirmedAt  sql.NullTime `db:"confirmed_at"`
+	ReferralCode string       `db:"referral_code"`
+}
+
+// ReferralStatus represents a referral workflow status: registered -> installed -> confirmed
+type ReferralStatus string
+
+const (
+	// RegisteredReferralStatus means the receiver registered with the sender referral code.
+	RegisteredReferralStatus ReferralStatus = "registered"
+	// InstalledReferralStatus means the receiver installed the Browser and restored the account with their seed.
+	InstalledReferralStatus ReferralStatus = "installed"
+	// ConfirmedReferralStatus means the reward has been sent to the sender and receiver.
+	ConfirmedReferralStatus ReferralStatus = "confirmed"
+)
+
+// Referral ...
+type Referral struct {
+	Owner          string         `db:"sender"`
+	Receiver       string         `db:"receiver"`
+	Status         ReferralStatus `db:"status"`
+	RegisteredAt   time.Time      `db:"registered_at"`
+	InstalledAt    sql.NullTime   `db:"installed_at"`
+	ConfirmedAt    sql.NullTime   `db:"confirmed_at"`
+	SenderReward   sql.NullInt32  `db:"sender_reward"`
+	ReceiverReward sql.NullInt32  `db:"receiver_reward"`
```

Comment:
It's better to use sdk.Dec

Reply:
Does it support Scan/Value form the sql package?

Reply:
not sure, but I believe it does

Reply:
checked, it does not.